app/src/main/java/com/blogspot/e_kanivets/moneytracker/activity/record/MainActivity.kt

```diff
@@ -134,17 +137,43 @@ class MainActivity : BaseDrawerActivity() {
     override fun update() {
         recordList = recordController.getRecordsForPeriod(period)
         recordList = recordList.reversed()
+        updateRecordAdapterDataList()
 
         val currency = currencyController.readDefaultCurrency()
 
         val reportMaker = ReportMaker(rateController)
         val report = reportMaker.getRecordReport(currency, period, recordList)
 
-        recordAdapter.setRecords(recordList, report, currency, reportMaker.currencyNeeded(currency, recordList))
+        recordAdapter.setRecords(recordAdapterDataList, report, currency, reportMaker.currencyNeeded(currency, recordList))
 
         fillDefaultAccount()
     }
 
+    private fun getCountHeadersItems(position: Int): Int {
```

Comment:
Smart, but I would take one more look on the places where we use `getCountHeadersItems`. Probably we can get a record without this magic :)